src/libgap-api.c

```diff
@@ -0,0 +1,337 @@
+// LibGAP API - API for using GAP as shared library.
+
+#include "libgap-api.h"
+
+#include "bool.h"
+#include "calls.h"
+#include "gapstate.h"
+#include "gvars.h"
+#include "intobj.h"
+#include "opers.h"
+#include "plist.h"
+#include "stringobj.h"
+
+Obj FuncREAD_ALL_COMMANDS(
+    Obj self, Obj instream, Obj echo, Obj capture, Obj resultCallback);
+
+// Setup and initialisation
+void GAP_Initialize(int          argc,
+                    char **      argv,
+                    char **      env,
+                    CallbackFunc markBagsCallback,
+                    CallbackFunc errorCallback)
+{
+    InitializeGap(&argc, argv, env);
+    SetExtraMarkFuncBags(markBagsCallback);
+    STATE(JumpToCatchCallback) = errorCallback;
+}
+
+// TODO: This function is far from completely finalizing a GAP session
+//       it potentially has to close all open file descriptors, kill
+//       subprocesses, etc
+void GAP_Finalize(void)
+{
+    FinishBags();
+}
+
+void GAP_ClearError(void)
+{
+    ClearError();
+}
+
+void GAP_Collect(void)
+{
+    CollectBags(0, 1);
+}
+
+const char * GAP_GetTNam(Obj val)
+{
+    return TNAM_OBJ(val);
+}
+
+uint64_t GAP_GetTNum(Obj val)
+{
+    return TNUM_OBJ(val);
+}
+
+
+//
+// Booleans
+//
+bool GAP_IsBoolean(Obj val)
+{
+    return TNUM_OBJ(val) == T_BOOL;
+}
+
+// What about fail :/
+Obj GAP_MakeBoolean(bool val)
+{
+    return val == true ? True : False;
+}
+
+// Should the Get functions assert? get a pointer to put the return value
+// into?
+// If we return 0/1/2 then "Fail" becomes a "true" C value
+int GAP_GetBoolean(Obj val, int * res)
+{
+    GAP_ASSERT(res != NULL);
+    if (val == True)
+        *res = 1;
+    else if (val == False)
+        *res = 0;
+    else if (val == Fail)
+        *res = 2;
+    else
+        *res = -1;
+}
+
+inline Obj GAP_True()
+{
+    return True;
+}
+
+inline Obj GAP_False()
+{
+    return False;
+}
+
+inline Obj GAP_Fail()
+{
+    return Fail;
+}
+
+
+// TODO: depending on 32/64bit? Int?
+bool GAP_IsImmediateInteger(Obj val)
+{
+    return IS_INTOBJ(val);
+}
+
+Obj GAP_MakeImmediateInteger(int64_t val)
+{
+    INTOBJ_INT(val);
```

Comment:
This is missing a return; but worth, it can be easily misused, namely if the input does not fit into an immediate object. Also, why hardcode 64 bit API here? Why not use `Int val` as argument?

But overall, I question the usefulness of this function. I think it would be much better to expose these functions suitably:
```
extern Obj ObjInt_Int(Int i);
extern Obj ObjInt_UInt(UInt i);
extern Obj ObjInt_Int8(Int8 i);
extern Obj ObjInt_UInt8(UInt8 i);

extern Int Int_ObjInt(Obj i);    
extern UInt UInt_ObjInt(Obj i);    
extern Int8 Int8_ObjInt(Obj i);    
extern UInt8 UInt8_ObjInt(Obj i);    
```